app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,130 @@
+class TweetController < ApplicationController
+    get '/tweets' do
+        if !logged_in?
+            redirect to '/login'
+        end
+
+        @tweets = Tweet.all
+
+        erb :'/tweets/index'
+    end
+
+    post '/tweets' do
+        if !logged_in?
+            redirect to '/login'
+        end
+
+        if params[:content].empty?
+            session[:new_tweet_error] = true
+            session[:new_tweet_error_message] = 'Tweets must not be blank!'
+            redirect to '/tweets/new'
+        end
+
+        tweet = Tweet.create(content: params[:content])
+        tweet.user = user

```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```